client/src/pages/Signup/signup.jsx

```diff
@@ -3,293 +3,211 @@ import { Link } from "react-router-dom";
 import toast from "react-hot-toast";
 import axios from "axios";
 import { Navigate } from "react-router-dom";
+import "./Signup.css"; // Import the custom CSS file for additional styles
 
 const Signup = () => {
-    const [name, setName] = useState("");
-    const [email, setEmail] = useState("");
-    const [passwd, setPasswd] = useState("");
-    const [ppic, setPpic] = useState("");
-    const [street, setStreet] = useState("");
-    const [city, setCity] = useState("");
-    const [state, setState] = useState("");
-    const [pincode, setPincode] = useState(0);
-    const [registered, setRegistered] = useState(false);
-    const registerf = async (e) => {
-        e.preventDefault();
-        if (
-            !(
-                name.length > 0 &&
-                email.length > 0 &&
-                passwd.length > 0 &&
-                street.length > 0 &&
-                city.length > 0 &&
-                state.length > 0 &&
-                pincode > 0
-            )
-        ) {
-            toast.error("All fields are necessary");
-            return;
-        }
-        try {
-            const { data } = await axios.post(
-                "http://localhost:3000/api/users/signup",
-                {
-                    name,
-                    email,
-                    ppic,
-                    passwd,
-                    street,
-                    city,
-                    state,
-                    pincode,
-                },
-                {
-                    headers: {
-                        "Content-Type": "application/json",
-                    },
-                    withCredentials: true,
-                }
-            );
-            toast(data.message);
-            setRegistered(true);
-        } catch (error) {
-            toast.error(error.response.data.message);
-            console.error(error);
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [passwd, setPasswd] = useState("");
+  const [ppic, setPpic] = useState("");
+  const [street, setStreet] = useState("");
+  const [city, setCity] = useState("");
+  const [state, setState] = useState("");
+  const [pincode, setPincode] = useState("");
+  const [rememberMe, setRememberMe] = useState(false);
+  const [registered, setRegistered] = useState(false);
+
+  const registerf = async (e) => {
+    e.preventDefault();
+    if (
+      !(
+        name.length > 0 &&
+        email.length > 0 &&
+        passwd.length > 0 &&
+        street.length > 0 &&
+        city.length > 0 &&
+        state.length > 0 &&
+        pincode.length > 0
+      )
+    ) {
+      toast.error("All fields are necessary");
```

Comment:
Please look for a better approach than checking the length. (empty functions etc.)